### PR TITLE
[Asset] Custom settings shouldn't go to cache when exceeding a certain size

### PR DESCRIPTION
**Pull request by brusch**
Alternative approach to #16240

### CODE REVIEW

File: models/Asset.php [hunk 1]
```diff
@@ -1254,6 +1294,10 @@ public function getCustomSettings(): array
     public function setCustomSettings(mixed $customSettings): static
     {
         if (is_string($customSettings)) {
+            if(strlen($customSettings) > 10e6) {
```
Comment (BlackbitDevs) [inline models/Asset.php:1297]: This does not get triggered if 
```php
$asset->setCustomSettings(['long' => str_repeat('x', 10e7)]);
```
gets used, does it?

Even more important, this does not get triggered if https://github.com/pimcore/pimcore/blob/b3cda742757100ee66f3f8093b2789a7c1eebe09/models/Asset/MetaData/EmbeddedMetaDataTrait.php#L88 sets the asset metadata as `customSettings` - which was the actual problem of #16240

Comment (brusch) [reply models/Asset.php:1297]: @BlackbitDevs it does, when it's setting the data from database, either here: 
https://github.com/pimcore/pimcore/blob/6f976580522eacbb293ea5d02732d93d1af036d5/models/Asset.php#L1257
or here: 
https://github.com/pimcore/pimcore/blob/9bff50b5ce8d207e07ff22090d266f76888329aa/models/Asset/Dao.php#L54

And when saving the asset, it doesn't go to cache anyway within the same process, so it needs to be loaded from database at least once in a different process after saving before getting stored in the cache 😉

Comment (BlackbitDevs) [reply models/Asset.php:1297]: > And when saving the asset, it doesn't go to cache anyway within the same process, so it needs to be loaded from database at least once in a different process after saving before getting stored in the cache

Yes you are right, it gets added to the cache when the asset gets loaded. And then it has been populated in https://github.com/pimcore/pimcore/blob/9bff50b5ce8d207e07ff22090d266f76888329aa/models/Asset/Dao.php#L54

**Comment by BlackbitDevs** [comment]
Ok, looks good to me @brusch

**Pull request merged**